doc/filters/merge.rst

```diff
@@ -42,6 +42,25 @@ overridden.
         
 .. note::
 
-    Internally, Twig uses the PHP `array_merge`_ function.
+    Internally, Twig uses the PHP `array_merge`_ function, which means that it will loose any numerical index after a merge, unless the numerical index is concatenated with something to make a string.
```

Comment:
As we already link to the array_merge documentation, I don't think we need to duplicate part of the behavior here.